src/graph.c

```diff
@@ -54,6 +61,9 @@ void RAI_GraphFree(RAI_Graph* graph) {
   if (graph->backend == RAI_BACKEND_TENSORFLOW) {
     RAI_GraphFreeTF(graph);
   }
+  else if (graph->backend == RAI_BACKEND_TORCH) {
+    RAI_GraphFreeTorch(graph);
+  }
   else {
     // TODO: err properly
     printf("ERR: Unsupported backend.\n");
```

Comment:
maybe we should put an assert it cause this really should not happened right?

Reply:
You mean the unsupported backend, right? I agree

Reply:
(I got your comment now, see next response to comment)

Reply:
Also I would suggest to create a single graph struct with function pointers to Free, Run , .. and all the other capabilities. This way we can avoid all those if's and it will be easy to add more backends in the future ..

Reply:
Correct, we should definitely do some refactoring after this round of changes. I'm opening a dedicated issue now.